### ISSUE TITLE
[FormRecognizer] FieldValue should be able to handle 64-bit integers

### ISSUE THREAD

**Issue by kinelski** [Azure/azure-sdk-for-net#12021]
### Summary

The `Integer` value returned by the service to populate `FieldValue` can be as big as a 64-bit integer, but the client stores it as an `int`.

### Required CodeGen changes

Explicitly make `FieldValue_internal.valueInteger` a `long?` so CodeGen can parse it as a 64-bit integer.

Reference: [change a model property type](https://github.com/Azure/autorest.csharp#change-a-model-property-type)

### Required client changes

Rename `FieldValue.AsInt32` to `FieldValue.AsInt64` and update return type and docs description.

### Related

Currently there's an open discussion about defaulting `Integer` types to `long` in generated code: https://github.com/Azure/autorest.csharp/issues/744

**Comment by kinelski** [comment]
Getting this closed by #12320 since it's not blocked by the CodeGen discussion. I have opened a new issue to keep track of that (#12321) and added it to our backlog.
